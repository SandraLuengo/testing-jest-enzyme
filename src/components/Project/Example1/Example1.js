import React, { useState } from 'react';

const Example1 = ({ className }) => {
  const [inputState, toggleInputState] = useState(false);
  return (
    <div className={`${className} example1`}>
      <h2>Example 1: simple state change</h2>
      <button className="buttonClick" onClick={() => toggleInputState(true)}>
        Change State{' '}
      </button>
      <p className="text">{`inputState: ${inputState}`}</p>
    </div>
  );
};

export default Example1;
