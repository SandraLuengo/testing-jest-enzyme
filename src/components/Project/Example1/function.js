const desmonto = () => 'unmount';

export default desmonto;
