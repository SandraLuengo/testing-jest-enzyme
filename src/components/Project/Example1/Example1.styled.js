import styled from 'styled-components';
import { connect } from 'react-redux';
import Example1 from './Example1';

const mapStateToProps = state => ({
  toggleColor: state.changeColorReducer.toggleColor,

});


export default connect(mapStateToProps)(styled(Example1).attrs({})`
    &.example1{
        transition: background-color .5s ease-in-out;
        background-color:${({ toggleColor }) => (toggleColor ? 'red' : 'blue')};
    }
`);
