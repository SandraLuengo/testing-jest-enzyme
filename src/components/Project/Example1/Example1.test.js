import React from 'react';
import { shallow, mount } from 'enzyme';
import Example1 from './Example1';

describe('<Example1/>', () => {
  const wrapper = mount(<Example1 />);

  it('renders correctly', () => {
    shallow(<Example1 />);
  });
  it('exist an element with class example1', () => {
    expect(wrapper.find('.example1').length).toBe(1);
  });
  it('expected that click on button change the text', () => {
    wrapper.find('.buttonClick').simulate('click');
    const pText = wrapper.find('.text');
    expect(pText.text()).toBe('inputState: true');
  });
});
