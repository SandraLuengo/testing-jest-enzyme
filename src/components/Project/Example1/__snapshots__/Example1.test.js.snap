// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing styled Example1 snapshot should change 1`] = `ShallowWrapper {}`;

exports[`testing styled Example1 snapshot should change 2`] = `ShallowWrapper {}`;
